File: src/main/java/com/procurement/contracting/model/dto/Can.java
package com.procurement.contracting.model.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.procurement.contracting.model.dto.ocds.Contract;
import javax.validation.constraints.NotNull;
import lombok.Getter;

@Getter
public class Can {

    @NotNull
    @JsonProperty("token")
    private final String token;

    @NotNull
    @JsonProperty("contract")
    private final Contract contract;

    public Can(@JsonProperty("token") final String token,
               @JsonProperty("contract") final Contract contract) {
        this.token = token;
        this.contract = contract;
    }
}

File: src/main/kotlin/com/procurement/contracting/model/dto/Can.kt
package com.procurement.contracting.model.dto

import com.fasterxml.jackson.annotation.JsonProperty
import com.procurement.contracting.model.dto.ocds.Contract
import javax.validation.Valid
import javax.validation.constraints.NotNull

data class Can(

        @JsonProperty("token") @NotNull
        val token: String,

        @JsonProperty("contract") @Valid @NotNull
        val contract: Contract
)
